### DISCUSSION TITLE
Use `.pkgx.yaml` in Dockerfile?

### DISCUSSION THREAD

**Discussion by gaggle** [pkgxdev/pkgx#775] [Help] [answered]
Is there a way to use the `.pkgx.yaml` file inside a Dockerfile, so its environment becomes synchronized with the local dev environment? I have a project I'd like to containerize, and I figured I could maybe do something like this:

```Dockerfile
FROM pkgxdev/pkgx
WORKDIR /app
COPY .pkgx.yaml .
RUN dev
```

But that actually outputs:

```sh
$ docker build --progress=plain .
#8 [build 4/4] RUN dev
#8 sha256:a66705348c4597320598f745464945e2efab1513b3a85c1d9697ae7963eb30a6
#8 0.181 /bin/sh: 1: dev: not found
#8 ERROR: executor failed running [/bin/sh -c dev]: exit code: 127
------
 > [build 4/4] RUN dev:
------
```

Am I abusing/misunderstanding how pkgx wants to work with a container? The [documentation page](https://docs.pkgx.sh/run-anywhere/docker) seems to suggest we can build our own dependencies using `RUN env +node@16`, but then I'd be replicating my `.pkgx.yaml` dependencies so I figured maybe the `dev` command would/should work?

**Comment by jhheider** [comment]
Likely a consequence of this: https://github.com/pkgxdev/pkgx/issues/769#issuecomment-1747457656.
Maybe:
```docker
RUN eval "$(pkgx --shellcode)" && dev
```
?

**Reply by mxcl** [reply, marked as answer]
yes, typing `dev` will work with the next release 1.0.1 (today I hope).

In the meantime the shellcode for dev is `eval "$(pkgx --internal.activate $PWD)"` if you like, edit or yes @jhheider’s example is the endorsed route.
